strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/storage/indexing/remote/RepositoryProxyIndexFounder.java

```diff
@@ -0,0 +1,127 @@
+
+import org.carlspring.strongbox.providers.io.RepositoryFiles;
+import org.carlspring.strongbox.providers.io.RepositoryPath;
+import org.carlspring.strongbox.providers.io.RepositoryPathResolver;
+import org.carlspring.strongbox.service.ProxyRepositoryConnectionPoolConfigurationService;
+import org.carlspring.strongbox.storage.indexing.*;
+import org.carlspring.strongbox.storage.indexing.RepositoryIndexDirectoryPathResolver.RepositoryIndexDirectoryPathResolverQualifier;
+import org.carlspring.strongbox.storage.indexing.RepositoryIndexFounder.RepositoryIndexFounderQualifier;
+import org.carlspring.strongbox.storage.indexing.RepositoryIndexingContextFactory.RepositoryIndexingContextFactoryQualifier;
+import org.carlspring.strongbox.storage.repository.Repository;
+import org.carlspring.strongbox.storage.repository.RepositoryTypeEnum;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.Date;
+import java.util.Objects;
+
+import org.apache.maven.index.incremental.DefaultIncrementalHandler;
+import org.apache.maven.index.updater.DefaultIndexUpdater;
+import org.apache.maven.index.updater.IndexUpdateRequest;
+import org.apache.maven.index.updater.IndexUpdateResult;
+import org.apache.maven.index.updater.IndexUpdater;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author carlspring
+ * @author Przemyslaw Fusik
+ */
+@Component
+@RepositoryIndexFounderQualifier(RepositoryTypeEnum.PROXY)
+public class RepositoryProxyIndexFounder
+        extends AbstractRepositoryIndexFounder
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(RepositoryProxyIndexFounder.class);
+
+    private final IndexUpdater indexUpdater = new DefaultIndexUpdater(new DefaultIncrementalHandler(), null);
+
+    @Inject
+    private ProxyRepositoryConnectionPoolConfigurationService proxyRepositoryConnectionPoolConfigurationService;
+
+    @Inject
+    private ResourceFetcherFactory resourceFetcherFactory;
+
+    @Inject
+    private RepositoryPathResolver repositoryPathResolver;
+
+    @Inject
+    @RepositoryIndexingContextFactoryQualifier(IndexTypeEnum.REMOTE)
+    private RepositoryIndexingContextFactory indexingContextFactory;
+
+    @Inject
+    @RepositoryIndexDirectoryPathResolverQualifier(IndexTypeEnum.REMOTE)
+    private RepositoryIndexDirectoryPathResolver indexDirectoryPathResolver;
+
+    @Override
+    protected void onIndexingContextCreated(final RepositoryPath repositoryIndexDirectoryPath,
+                                            final RepositoryCloseableIndexingContext indexingContext)
+            throws IOException
+    {
+        final Repository repository = indexingContext.getRepositoryRaw();
+
+        final String storageId = repository.getStorage().getId();
+        final String repositoryId = repository.getId();
+
+        final IndexUpdateResult updateResult = fetchIndex(indexingContext, repository, storageId, repositoryId);
```

Comment:
@fuss86 am I understand correctly that for proxy repositories the index will be fetched from remote repository and there will be no appropriate `ArtifactEntries` and `ArtifactGroups`?

Reply:
yes, 
- `proxy` repositories will have only `remote` index
- `hosted` repositories will have only `local` index created from `ArtifactEntry`
- `group` repositories will have only `local` index merged from underlying repositories

Reply:
@fuss86 can we parse remote index and store it as `ArtifactEntries`? I mean that all the other layouts works like this, they parse remote matadata and store it into our DB, so we probably should do the same with Maven. 

@carlspring can you share your thoughts as well?

Reply:
> @fuss86 can we parse remote index and store it as ArtifactEntries

yes, we can

> @carlspring can you share your thoughts as well?

I think it would increase search hits even without having artifacts physically stored in the FS. I think it's good idea, but it also deserves being done in a follow-up task ;)